vm/qemu/qemu.go

```diff
@@ -750,6 +754,54 @@ func (inst *instance) Diagnose(rep *report.Report) ([]byte, bool) {
 	return ret, false
 }
 
+func needsRegisterInfo(rep *report.Report) bool {
+	switch rep.Type {
+	case crash.NullPtrDerefBUG,
+		crash.KASANNullPtrDerefRead,
+		crash.KASANNullPtrDerefWrite,
+		crash.KFENCEMemoryCorruption:
+		return true
+	case crash.KASANRead,
+		crash.KASANWrite,
+		crash.KASANUseAfterFreeRead,
+		crash.KASANUseAfterFreeWrite,
+		crash.KFENCERead,
+		crash.KFENCEWrite,
+		crash.KFENCEUseAfterFreeRead,
+		crash.KFENCEUseAfterFreeWrite:
+		return true
+	case crash.KASANInvalidFree,
+		crash.KFENCEInvalidFree:
+		return true
+	case crash.MemorySafetyBUG,
+		crash.MemorySafetyUBSAN:
+		return true
+	case crash.Bug:
+		return true
+	case crash.Warning,
+		crash.AtomicSleep,
+		crash.Hang,
+		crash.DoS,
+		crash.KCSANAssert,
+		crash.KCSANDataRace,
+		crash.KCSANUnknown,
```

Comment:
@melver  we currently don't have a qemu+KCSAN instance, but if we had, would it be helpful to have a register dump after the moment of a crash?

See an example of such output at the bottom of this console output log:
https://syzkaller.appspot.com/text?tag=CrashLog&x=1511e762580000